src/services/utils.ts

```diff
@@ -60,59 +67,44 @@ const downloadFile = (blob: Blob, filename: string, type?: FileType) => {
     document.body.removeChild(link);
 };
 
-const prepareRequest = (init: any, token?: string) => {
+function prepareRequest(init?: InitRequest, token?: Token): RequestInit {
     if (!(typeof init === 'undefined' || typeof init === 'object')) {
         throw new TypeError(
-            'First argument of prepareRequest is not an object : ' + typeof init
+            `Argument 2 of backendFetch is not an object ${typeof init}`
         );
     }
-    const initCopy = Object.assign({}, init);
+    const initCopy: RequestInit = Object.assign({}, init);
     initCopy.headers = new Headers(initCopy.headers || {});
     const tokenCopy = token ?? getUserToken();
     initCopy.headers.append('Authorization', 'Bearer ' + tokenCopy);
     return initCopy;
-};
+}
 
-const safeFetch = (url: string, initCopy: any) => {
+function safeFetch<T>(url: string, initCopy: any): Promise<T> {
     return fetch(url, initCopy).then((response) =>
         response.ok ? response : handleError(response)
-    );
-};
+    ) as Promise<T>;
+}
 
-const handleError = (response: any) => {
+function handleError(response: Response): Promise<never> {
     return response.text().then((text: string) => {
         const errorName = 'HttpResponseError : ';
+        let error: ErrorWithStatus;
         const errorJson = parseError(text);
-        let customError: Error & { status?: string };
-        if (
-            errorJson &&
-            errorJson.status &&
-            errorJson.error &&
-            errorJson.message
-        ) {
-            customError = new Error(
-                errorName +
-                    errorJson.status +
-                    ' ' +
-                    errorJson.error +
-                    ', message : ' +
-                    errorJson.message
-            );
-            customError.status = errorJson.status;
+        if (errorJson?.status && errorJson?.error && errorJson?.message) {
+            error = new Error(
+                `${errorName}${errorJson.status} ${errorJson.error}, message : ${errorJson.message}`
+            ) as ErrorWithStatus;
+            error.status = errorJson.status;
         } else {
-            customError = new Error(
-                errorName +
-                    response.status +
-                    ' ' +
-                    response.statusText +
-                    ', message : ' +
-                    text
```

Comment:
So we don't want to display this anymore ?

Reply:
That was because we consider that as we only fetch from our servers and we know that they return errors either as details json or juste http status, we don't have case where we can have raw text.

On top of that, if we get for example a webpage (which isn't json), we don't want to display uncontrolled content.